apps/web/app/api/sso/register/route.ts

```diff
@@ -0,0 +1,68 @@
+import { NextResponse } from "next/server";
+import { z } from "zod";
+import { isAdmin } from "@/utils/admin";
+import { auth } from "@/utils/auth";
+import { createScopedLogger } from "@/utils/logger";
+import { withAuth } from "@/utils/middleware";
+import { registerSSOProvider } from "@/utils/register-sso-provider";
+
+const logger = createScopedLogger("api/sso/register");
+
+const registerSSOProviderBody = z.object({
+  idpMetadata: z
+    .string()
+    .min(1, "IdP metadata is the SAML IDP XML metadata of the provider"),
+  providerId: z
+    .string()
+    .min(
+      1,
+      "A provider ID for internal usage. It must match the callback URL api/auth/sso/saml2/callback/<providerId>",
+    ),
+  organizationName: z.string().min(1, "The organization name of the provider"),
+  domain: z.string().min(1, "The domain of the provider"),
+});
+
+export const POST = withAuth(async (request) => {
+  try {
+    // Only admins should be able to register SSO providers
+    const session = await auth();
+    if (!isAdmin({ email: session?.user?.email })) {
+      return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+    }
```

Comment:
Fine for anyone to create sso. Is there a reason not to? Other than it being a premium feature.